File: src/css/lib/vanillabox.rb
module Sass::Script::Functions
	def high_res(url, ratio)
		assert_type url, :String, :url
		assert_type ratio, :Number, :ratio
 
		str = url.value
		index = str.rindex('.')
		if index == nil
			index = str.length()
		end

		high_res_str = str.insert(index, "@#{ratio.value}x")
		return Sass::Script::String.new(high_res_str, :String)
	end
	declare :high_res, [:url, :ratio]
end

File: src/css/lib/vanillabox.js
const sass = require('node-sass');

module.exports = {
	'high_res($url, $ratio)': function(url, ratio) {
		const str = url.getValue();
		let index = str.lastIndexOf('.');
		if (index < 0) {
			index = str.length();
		}

		const highResStr = str.slice(0, index) +
			`@${ratio.getValue()}x` +
				str.slice(index);
		return sass.types.String(highResStr);
	}
};
